numba/typed/typedlist.py

```diff
@@ -397,7 +397,7 @@ def __str__(self):
 
     def __repr__(self):
         body = str(self)
-        prefix = str(self._list_type)
+        prefix = str(self._list_type) if self._typed else "ListType[UNTYPED]"
```

Comment:
Perhaps a `ListType[Undefined]` ? As in, we know it's a `ListType` but the item type is `Undefined`, I think this makes it more explicit and consistent with other things (which tend to refer to "undefined")?

Reply:
Fixed